core/src/main/java/org/infinispan/container/offheap/MemoryAddressHash.java

```diff
@@ -16,11 +16,13 @@
 
    private final long memory;
    private final int pointerCount;
+   private final OffHeapMemoryAllocator allocator;
 
-   public MemoryAddressHash(int pointers) {
+   public MemoryAddressHash(int pointers, OffHeapMemoryAllocator allocator) {
       this.pointerCount = nextPowerOfTwo(pointers);
       long bytes = ((long) pointerCount) << 3;
-      memory = MEMORY.allocate(bytes);
+      this.allocator = allocator;
+      memory = allocator.allocate(bytes);
```

Comment:
I think we should really check all invocations to `allocate()`: if 0 is returned in means that something went wrong.

Reply:
Unsafe.cpp already throws an exception :)

http://hg.openjdk.java.net/jdk8/jdk8/hotspot/file/87ee5ee27509/src/share/vm/prims/unsafe.cpp#l595

Reply:
Damn, I read the implementation of os::malloc but then I forgot to unwind the stack :-)